core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -224,9 +224,10 @@ private object RTS {
 
     private final def collectDefect[E, A](e: ExitResult[E, A]): Option[List[Throwable]] =
       e match {
-        case ExitResult.Terminated(ts @ _ :: _) => Some(ts)
-        case ExitResult.Failed(_, ts @ _ :: _)  => Some(ts)
-        case _                                  => None
+        case ExitResult.Interrupted(_, ts @ _ :: _) => Some(ts)
+        case ExitResult.Terminated(t, ts)           => Some(t :: ts)
+        case ExitResult.Failed(_, ts @ _ :: _)      => Some(ts)
+        case _                                      => None
```

Comment:
I wonder if we should add `defects` as a method on `ExitResult` to cleanup this code and make it more useful outside `RTS`?